txredisapi.py

```diff
@@ -292,12 +310,18 @@ def connectionMade(self):
         self.factory.addConnection(self)
 
     def connectionLost(self, why):
+        # timeout: cleanup timeout watchdog
+        self.setTimeout(None)
         self.connected = 0
         self.script_hashes.clear()
         self.factory.delConnection(self)
         LineReceiver.connectionLost(self, why)
         while self.replyQueue.waiting:
-            self.replyReceived(ConnectionError("Lost connection"))
+            # timeout: raise TimeoutError on timeouts
+            if why == 'timeout':
+                self.replyReceived(TimeoutError("Lost connection"))
```

Comment:
If you raise `TimeoutError` where `ConnectionError` was raised before, may be it make sense to make `TimeoutError` subclass `ConnectionError` preserve compatibility?